q2_fmt/_peds.py

```diff
@@ -8,30 +8,191 @@
+    metadata, used_references = \
+        _check_associated_reference(reference_series, metadata, time_column,
+                                    filter_missing_references,
+                                    reference_column)
+    peds_df = pd.DataFrame(columns=['id', 'measure', 'recipients with feature',
+                                    'all possible recipients with feature',
+                                    'group', 'subject'])
+    for time, time_metadata in metadata.groupby(time_column):
+        peds_df = _compute_peds(peds_df=peds_df, peds_type="Feature",
+                                peds_time=time,
+                                reference_series=used_references, table=table,
+                                metadata=time_metadata,
+                                time_column=time_column,
+                                subject_column=subject_column,
+                                reference_column=reference_column)
+    return peds_df
+
+
+def _compute_peds(peds_df: pd.Series, peds_type: str, peds_time: int,
+                  reference_series: pd.Series, table: pd.Series,
+                  metadata: qiime2.Metadata, time_column: str,
+                  subject_column: str,
+                  reference_column: str) -> (pd.DataFrame):
+    table = table > 0
+    donordf = table[table.index.isin(reference_series)]
+    recipdf = _create_recipient_table(reference_series, metadata, table)
+
+    donormask = _create_masking(time_metadata=metadata, donordf=donordf,
+                                recipdf=recipdf,
+                                reference_column=reference_column)
+    maskedrecip = donormask & recipdf
+    if peds_type == "Sample":
+        num_sum = np.sum(maskedrecip, axis=1)
+        donor_sum = np.sum(donormask, axis=1)
+        for count, sample in enumerate(recipdf.index):
+            sample_row = metadata.loc[sample]
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
+                peds = num_sum[count] / donor_sum[count]
+
+            peds_df.loc[len(peds_df)] = [sample, peds, num_sum[count],
+                                         donor_sum[count],
+                                         sample_row[reference_column],
+                                         sample_row[subject_column],
+                                         sample_row[time_column]]
+        peds_df['id'].attrs.update({
+            'title': reference_series.index.name,
+            'description': 'Sample IDs'
+        })
+        peds_df['measure'].attrs.update({
+            'title': "PEDS",
+            'description': 'Proportional Engraftment of Donor Strains '
+        })
+        peds_df['group'].attrs.update({
+            'title': time_column,
+            'description': 'Time'
+        })
+        peds_df["subject"].attrs.update({
+            'title': subject_column,
+            'description': 'ID to link samples across time'
```

Comment:
```suggestion
            'description': 'Subject IDs linking samples across time'
```
This seemed more clear to me, but feel free to take it or leave it 🤷‍♀️